Dockerfile

```diff
@@ -0,0 +1,24 @@
+  # STAGE 1
+  FROM golang:1.13.4 as builder
+
+  WORKDIR /go/github.com/jetstack/preflight
+
+  # Run a dependency resolve with just the go mod files present for
+  # better caching
+  COPY ./go.mod .
+  COPY ./go.sum .
+
+  RUN go mod download
+
+  ## Bring in everything else and build an amd64 image
+  COPY . .
+  RUN GOOS=linux GOARCH=amd64 go install .
+
+  # STAGE 2
+  # Use a distroless nonroot base image for just our executable
+  FROM gcr.io/distroless/base:nonroot
```

Comment:
Instead of using multi stage builds, which don't play well with go's module caching, consider using 'go_container.sh' from 'kind' which makes use of a docker volume to cache go dependencies, giving you host-like performance 😀 this works okay, but is just a bit slow for iterative development: https://github.com/kubernetes-sigs/kind/blob/master/hack/go_container.sh

Reply:
Will do in a separate PR: https://github.com/jetstack/preflight/issues/8